src/Bicep.Core/Semantics/FileSymbol.cs

```diff
@@ -248,6 +248,11 @@ public override void Accept(SymbolVisitor visitor)
                 return new(DiagnosticBuilder.ForDocumentStart().UsingDeclarationNotSpecified());
             }
 
+            if (usingDeclaration.Path is NoneLiteralSyntax)
+            {
+                return new(new InMemoryFileSemanticModel(string.Empty));
```

Comment:
I don't love this, but it may be the "least bad" option, compared to making the result nullable (which presumably is a pretty large refactor). I'd like to get some feedback from others on the team to see if we have any better suggestions /cc @shenglol, @jeskew.

Reply:
I need to dig into this further, but since we raise diagnostics when a params file supplies values for parameters that are not declared, don't we need to distinguish between cases where the author supplies `using none` vs when the target of the `using` statement is a real file that declares 0 parameters (of which `string.Empty` is an example).

Reply:
I'm fine with using a sentinel semantic model type if making the result nullable is too costly. In fact, it might be beneficial as it avoids null checks. However, I suggest renaming the type to EmptySemanticModel since all semantic models are technically in-memory.

Regarding Jonny's point, it appears that `SemanticModel.GatherParameterMismatchDiagnostics` has been updated to ensure that `using none` won't trigger any parameter assignment or declaration mismatch errors. However, I'm not sure if there are missing cases where special handling is required. I suggest adding some tests to ensure that we distinguish between the two cases properly.